app/models/products/user_note_mode.rb

```diff
@@ -0,0 +1,49 @@
+module Products
+
+  class UserNoteMode
+
+    attr_reader :raw_value
+
+    def self.values
+      valid_values.map { |value| new(value) }
+    end
+
+    def self.valid_values
+      %w(hidden optional required).freeze
+    end
+
+    def initialize(raw_value)
+      @raw_value = raw_value
+      freeze
```

Comment:
I wanted to do that, but then we get an exception rather than a validation error when assigning within the Product. If anyone has a good suggestion for how best to balance that, I'm open.